lnbits/extensions/boltz/tasks.py

```diff
@@ -0,0 +1,154 @@
+                    )
+                    await update_swap_status(swap.id, "complete")
+                else:
+                    if swap_status.hit_timeout:
+                        if not swap_status.has_lockup:
+                            logger.warning(
+                                f"Boltz - swap: {swap.id} hit timeout, but no lockup tx..."
+                            )
+                            await update_swap_status(swap.id, "timeout")
+                        else:
+                            logger.debug(f"Boltz - refunding swap: {swap.id}...")
+                            await create_refund_tx(swap)
+                            await update_swap_status(swap.id, "refunded")
+
+            except Exception as exc:
+                logger.error(f"Boltz - swap: {swap.id} - {str(exc)}")
+
+    if len(reverse_swaps) > 0:
+        logger.debug(f"Boltz - {len(reverse_swaps)} pending reverse swaps")
+        for reverse_swap in reverse_swaps:
+            try:
+                swap_status = get_swap_status(reverse_swap)
+
+                if swap_status.exists is False:
+                    logger.debug(
+                        f"Boltz - reverse_swap: {reverse_swap.boltz_id} does not exist."
+                    )
+                    await update_swap_status(reverse_swap.id, "failed")
+                    continue
+
+                # if timeout hit, boltz would have already refunded
+                if swap_status.hit_timeout:
+                    logger.debug(
+                        f"Boltz - reverse_swap: {reverse_swap.boltz_id} timeout."
+                    )
+                    await update_swap_status(reverse_swap.id, "timeout")
+                    continue
+
+                if not swap_status.has_lockup:
+                    # start listener for onchain address
+                    logger.debug(
+                        f"Boltz - reverse_swap: {reverse_swap.boltz_id} restarted onchain address listener."
+                    )
+                    await start_onchain_listener(reverse_swap)
+                    continue
+
+                if reverse_swap.instant_settlement or swap_status.confirmed:
+                    await create_claim_tx(reverse_swap, swap_status.lockup)
+                else:
+                    logger.debug(
+                        f"Boltz - reverse_swap: {reverse_swap.boltz_id} restarted confirmation listener."
+                    )
+                    await start_confirmation_listener(reverse_swap, swap_status.lockup)
+
+            except Exception as exc:
+                logger.error(f"Boltz - reverse swap: {reverse_swap.id} - {str(exc)}")
+
+
+async def wait_for_paid_invoices():
+    invoice_queue = asyncio.Queue()
+    register_invoice_listener(invoice_queue)
+
+    while True:
+        payment = await invoice_queue.get()
+        await on_invoice_paid(payment)
+
+
+async def on_invoice_paid(payment: Payment) -> None:
+    if "boltz" != payment.extra.get("tag"):
```

Comment:
A constant for `boltz` would be nice

Reply:
currently every extension does that. so for consistency i will keep that.